compiler/src/iree/compiler/Codegen/Common/GPU/GPUVectorDistribution.h

```diff
@@ -16,12 +16,14 @@
 
 namespace mlir::iree_compiler {
 
-/// A signature describing the layout for each operand of vector type for
-/// an operation.
-struct DistributionSignature {
-  SmallVector<VectorLayoutInterface> operands;
-  SmallVector<VectorLayoutInterface> results;
-};
+/// A signature describing the layout for each value of vector type which is
+/// an operand or result of this operation.
+///
+/// Two operands may be the same value, but since each value can only have
+/// one layout, we only need to keep track of the value, not the two operands
+/// separately.
+using DistributionSignature =
+    DenseMap<TypedValue<VectorType>, VectorLayoutInterface>;
```

Comment:
I know this might sound silly, but what if you had `vector.contract %0, %0, ...`, i.e. two of the same input. I assume layout conflicts have already been materialized at this point?

Reply:
+1. Shouldn't this be a map from `OpOperand` to layout interface? Or a SmallVector of pairs (or SmallMapVector) with placeholder values for non-vector operands/results?

Reply:
Yes, layout conflicts should have materialized already. A layout maps to each Value. Two OpOperands referring to the same value will always have the same layout.

Reply:
> +1. Shouldn't this be a map from `OpOperand` to layout interface? Or a SmallVector of pairs (or SmallMapVector) with placeholder values for non-vector operands/results?

No, layouts are always assigned to Value's not OpOperand.